lib/KTable/index.vue

```diff
@@ -0,0 +1,484 @@
+<template>
+
+  <div ref="tableWrapper" class="k-table-wrapper">
+    <table class="k-table" role="grid">
+      <caption v-if="caption">
+        {{ caption }}
+      </caption>
+      <thead>
+        <tr>
+          <th
+            v-for="(header, index) in headers"
+            :key="index"
+            tabindex="0"
+            :aria-sort="sortable && header.dataType !== DATA_TYPE_OTHERS ? getAriaSort(index) : null"
+            :class="{
+              [$computedClass(coreOutlineFocus)]: true,
+              sortable: sortable && header.dataType !== DATA_TYPE_OTHERS,
+              'sticky-header': true,
+              'sticky-column': index === 0,
+            }"
+            :style="[getHeaderStyle(header),
+                     sortKey === index ? { color: $themeBrand.primary.v_1000 } : { color: $themePalette.grey.v_800 },
+                     { backgroundColor: $themePalette.white } ,
+                     focusedColIndex === index ? { backgroundColor: $themePalette.grey.v_50 } : {}]"
+            role="columnheader"
+            data-focus="true"
+            :aria-colindex="index + 1"
+            @click="sortable && header.dataType !== DATA_TYPE_OTHERS ? handleSort(index) : null"
+            @keydown="handleKeydown($event, -1, index)"
+          >
+            <!--@slot Scoped slot for customizing the content of each header cell. Provides a header object `header` and its column index `colIndex`.-->
+
+            <slot name="header" :header="header" :index="index">
+              {{ header.label }}
+            </slot>
+            <span v-if="sortable && header.dataType !== DATA_TYPE_OTHERS" class="sort-icon">
+              <span v-if="sortKey === index && sortOrder === SORT_ORDER_ASC"><KIcon icon="dropup" :color="sortKey === index ? $themeBrand.primary.v_1100 : $themePalette.grey.v_800 " /></span>
+              <span v-else-if="sortKey === index && sortOrder === SORT_ORDER_DESC"><KIcon icon="dropdown" :color="sortKey === index ? $themeBrand.primary.v_1100 : $themePalette.grey.v_800 " /></span>
+              <span v-else><KIcon icon="sortColumn" :color="$themePalette.grey.v_800" /></span>
+            </span>
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr
+          v-for="(row, rowIndex) in finalRows"
+          :key="rowIndex"
+          :style="hoveredRowIndex === rowIndex || focusedRowIndex === rowIndex ? { backgroundColor: $themePalette.grey.v_50 } : {}"
```

Comment:
Would be better if we have this in a computed prop so its easier for people to understand the logic.

Reply:
done